src/commandsAndMenu.tsx

```diff
@@ -1571,6 +1584,37 @@ export async function showGitOperationDialog<T>(
   authentication?: Git.IAuth,
   retry = false
 ): Promise<string> {
+  /**
+   * Returns the current remote's URL based on the current remote name and all the remotes
+   */
+  async function getCurrentRemote(currentRemoteName: string): Promise<string> {
+    const remoteList = await model.getRemotes().then(remoteList => {
+      return remoteList;
+    });
+
+    const currentRemote = remoteList.find(
+      remoteURI => remoteURI.name === currentRemoteName
+    );
+
+    if (currentRemote) {
+      return currentRemote?.url;
+    } else {
+      // if the current remote isn't inside, what is best to return? Could maybe throw an error?
+      return '';
+    }
+  }
+
+  /**
+   * Returns the Git provider based on the domain name of the url
+   */
+  function checkUrlGitProvider(remoteUrl: string): string {
```

Comment:
We could rename this to match better the purpose of the function:

```suggestion
  function getGitProviderHost(remoteUrl: string): string {
```